### PR TITLE
[AIRFLOW-6014] - handle pods which are preempted and deleted by kuber…

### PR DESCRIPTION
**Pull request by atrbgithub**
…netes but not restarted

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-6014) issues and references them in the PR title. 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
This PR addresses the issue of when a pod is Preempted during the creation phase and due to pods having the following in the spec ```restartPolicy: Never``` The pod is never restarted and ends up as a queued task within Airflow until the scheduler is restarted.  

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unsure if it is possible to simulate this scenario. 

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by atrbgithub** [comment]
@dimberman would it be possible to take a look at this?

**Comment by kaxil** [comment]
cc @dimberman

File: airflow/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -334,10 +334,15 @@ def process_error(self, event: Any) -> str:
             (raw_object['reason'], raw_object['code'], raw_object['message'])
         )
 
+
     def process_status(self, pod_id: str, status: str, labels: Dict[str, str], resource_version: str) -> None:
```
Comment (inytar) [inline airflow/executors/kubernetes_executor.py]: `event` needs to be passed through here.

**Review by inytar — CHANGES_REQUESTED** [review]

File: airflow/executors/kubernetes_executor.py (hunk 1 above)
Comment (atrbgithub) [reply airflow/executors/kubernetes_executor.py]: @inytar thanks for this, I've pushed a commit to include it.

**Review by inytar — APPROVED** [review]
We are running a fork of Airflow with a similar fix, really hoping this gets merged on upstream!

**Review by dimberman — APPROVED** [review]

**Comment by dimberman** [comment]
LGTM. I put in a fix for a new namespace awareness and if tests pass I will merge. Apologies for the delay this one fell through the cracks :/

**Review by potiuk — APPROVED** [review]

**Pull request merged**